core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableFilter.java

```diff
@@ -68,4 +75,29 @@ public Result implement(EnumerableRelImplementor implementor, Prefer pref) {
     // EnumerableCalc is always better
     throw new UnsupportedOperationException();
   }
+
+  @Override public Pair<RelTraitSet, List<RelTraitSet>> passThroughTraits(
+      RelTraitSet required) {
+    RelCollation collation = required.getCollation();
+    if (collation == null || collation.getFieldCollations().size() == 0) {
+      return null;
+    }
+
+    // Filter, not like Project, does not change on which column that collation defines.
+    return Pair.of(required, ImmutableList.of(required));
+  }
+
+  @Override public Pair<RelTraitSet, List<RelTraitSet>> deriveTraits(
+      final RelTraitSet childTraits, final int childId) {
+    RelCollation collation = childTraits.getCollation();
+    if (collation == null || collation.getFieldCollations().size() == 0) {
+      return null;
+    }
+
+    return Pair.of(getTraitSet().replace(collation), ImmutableList.of(childTraits));
+  }
+
+  @Override public DeriveMode getDeriveMode() {
+    return DeriveMode.BOTH;
```

Comment:
why both? it has only 1 child, the default mode should be enough.

Reply:
Ah misunderstood what this mode controls. Agreed the default mode is enough.